src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -4764,12 +4764,30 @@ private void AddSynthesizedTupleMembersIfNecessary(MembersAndInitializersBuilder
                         }
                         break;
 
+                    case SyntaxKind.IncompleteMember:
+                        {
+                            var incompleteMemberSyntax = (IncompleteMemberSyntax)m;
+                            if (incompleteMemberSyntax.Type is not null)
+                            {
+                                // We want to produce binding errors for incomplete members
+                                _ = bodyBinder.BindType(incompleteMemberSyntax.Type, diagnostics);
+                            }
+
+                            foreach (var attributeList in incompleteMemberSyntax.AttributeLists)
+                            {
+                                foreach (var attribute in attributeList.Attributes)
+                                {
+                                    bodyBinder.BindType(attribute.Name, diagnostics);
+                                }
+                            }
+                        }
+                        break;
+
```

Comment:
Note to self: This doesn't handle attribute arguments (e.g, `[MyAttribute(typeof(...))]`)